### PR TITLE
chore: remove TS SwitchBoard class + purge navRef pattern

### PR DESCRIPTION
**Pull request by ds300**
### Description

Follow up from #3771 

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have added tests for my changes, or my changes don't require testing, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/master/docs/adding_state_migrations.md))
- [x] I have added a [CHANGELOG.yml](/CHANGELOG.yml) entry or my changes do not require one.

### CODE REVIEW

File: src/lib/utils/renderMarkdown.tsx [hunk 1]
```diff
@@ -46,10 +46,7 @@ export function defaultRules({
     link: {
       react: (node, output, state) => {
         state.withinText = true
-        // @ts-expect-error STRICTNESS_MIGRATION --- 🚨 Unsafe legacy code 🚨 Please delete this and fix any type errors if you have time 🙏
```
Comment (ashfurrow) [inline src/lib/utils/renderMarkdown.tsx:49]: So many deleted strictness migration comments 💆

**Comment by ashfurrow** [comment]
I haven't done a comprehensive review of all changes (I'll leave that to Mounir) but I've reviewed enough to get the gist. Really nice cleanup, so many fewer `navRef`s!

**Review by MounirDhahri — APPROVED** [review]

**Comment by MounirDhahri** [comment]
🕺🕺🕺 one step closer in the right direction 🎉

**Comment by MounirDhahri** [comment]
#squashongreen

**Pull request merged**